### PR TITLE
NIFI 3863 - Add LookupRecord, PartitionRecord processors

### PR DESCRIPTION
**Pull request by markap14**
I still need to add some more documentation and unit tests but wanted to go ahead and get this up so that if anyone has a chance to review or test they can do so.

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by joewitt** [comment]
@markap14  i think you dont want this in the PR
 create mode 100644 nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/hs_err_pid26624.log

Can you please squash when it is time for a formal review.  I am going through this already though.  This is cool stuff!

**Comment by joewitt** [comment]
I think for LookupAttribute it should either send all events to success (so it doesn't change/partition the stream other than to possibly enrich each record).  

I think if the lookup recordpath does not find anything the enrichment should just move on just like if the result record is empty.

**Comment by joewitt** [comment]
this thing is insanely useful.  I will post my template when done reviewing.  But this is supremely cool.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PartitionRecord.java [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+    description="Each dynamic property represents a RecordPath that will be evaluated against each record in an incoming FlowFile. When the value of the RecordPath is determined "
+        + "for a Record, an attribute is added to the outgoing FlowFile. The name of the attribute is the same as the name of this property. The value of the attribute is the same as "
+        + "the value of the field in the Record that the RecordPath points to. Note that no attribute will be added if the value returned for the RecordPath is null or is not a scalar "
+        + "value (i.e., the value is an Array, Map, or Record).",
+    supportsExpressionLanguage=true)
+@WritesAttributes({
+    @WritesAttribute(attribute="record.count", description="The number of records in an outgoing FlowFile"),
+    @WritesAttribute(attribute="mime.type", description="The MIME Type that the configured Record Writer indicates is appropriate"),
+    @WritesAttribute(attribute="<dynamic property name>",
+        description="For each dynamnic property that is added, an attribute may be added to the FlowFile. See the description for Dynamic Properties for more information.")
```
Comment (mattyb149) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PartitionRecord.java]: Minor typo, "dynamnic" should be "dynamic"

Comment (markap14) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PartitionRecord.java]: Thanks. Addressed.

File: nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/record/script/ScriptedRecordSetWriter.java [hunk 2]
```diff
@@ -131,6 +129,7 @@ protected boolean reloadScript(final String scriptBody) {
             }
 
         } catch (final Exception ex) {
+            ex.printStackTrace();
```
Comment (joewitt) [inline nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/record/script/ScriptedRecordSetWriter.java]: removing this and addressing a couple checkstyle issues

**Pull request closed without merging**